httpbakery/client.go

```diff
@@ -428,9 +428,10 @@ func (c *Client) interact(location, visitURLStr, waitURLStr string) (*macaroon.M
 		if err := json.NewDecoder(waitResp.Body).Decode(&resp); err != nil {
 			return nil, errgo.Notef(err, "cannot unmarshal wait error response")
 		}
-		return nil, errgo.WithCausef(nil, &DischargeError{
+		dischargeErr := &DischargeError{
```

Comment:
As this could also be a message from apache it would be really helpful if it somehow included the body text if the response isn't some JSON.